setLocation.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"github.com/go-telegram-bot-api/telegram-bot-api"
+	"encoding/json"
+    "net/http"
+    "net/url"
+	"strconv"
+)
+
+type CepResponse struct {
+                            CEP        string  `json:"cep"`
+                            Logradouro string  `json:"logradouro"`
+                            Bairro     string  `json:"bairro"`
+                            Cidade     string  `json:"cidade"`
+                            Estado     string  `json:"estado"`
+                            Latitude   string  `json:"latitude"`
+                            Longitude  string  `json:"longitude"`
+                            Altitude   float64 `json:"altitude"`
+                            DDD        int     `json:"ddd"`
+                            IBGE       string  `json:"ibge"`
+                        }
+
+// locationHandler receive postal code from user.
+func locationHandler(update tgbotapi.Update, bot *tgbotapi.BotAPI, config botConfig, msgs botMessages) error {
+
+	isCep := false
+	pais := "??"
+	cep := ""
+	user := update.Message.From
+
+	if update.Message.Command() == "setlocation" {
+		args := strings.Split(strings.Trim(update.Message.CommandArguments(), " "), " ")
+
+		if len(args)!=2 {
+			sendReply(update.Message.Chat.ID, update.Message.MessageID, "/setlocation <pais> <código postal>", bot)
+			return nil
+		}
+
+		pais = args[0]
+		cep  = args[1]
+
+		if pais=="br" {
+			isCep=true
+		}
+	} else if update.Message.Command() == "cep" {
+		cep = strings.Trim(update.Message.CommandArguments(), " ")
+		isCep=true
+	} else {
+		log.Println("O que estou fazendo aqui comando -> "+update.Message.Command())
+		return nil
+	}
+
+	if (isCep) {
+		cep = strings.Replace(cep,"-","",-1)
+		cep = strings.Replace(cep,".","",-1)
+
+		err := procCep(user,cep,config)
+
+		if err == nil {
+			if isPrivateChat(update.Message.Chat) {
```

Comment:
Maybe you could return from this function earlier if it's not a PrivateChat? Also, note that the commands structure already has support to ignore non-private messages which are supposed to be private.